Add patch to support meson 0.59 which is the only version available in SLE 15 SP5
Author: Antonio Larrosa

---
 data/meson.build | 12 ++++++++++++
 meson.build      |  3 ++-
 po/meson.build   | 15 +++++++++++++++
 3 files changed, 29 insertions(+), 1 deletion(-)

diff --git a/data/meson.build b/data/meson.build
index 7ae7a9b..b920882 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -24,6 +24,17 @@ metainfo_with_relinfo = custom_target('gen-output',
     command : [ascli_exe, 'news-to-metainfo', '--limit=6', '@INPUT0@', '@INPUT1@', '@OUTPUT@']
 )
 
+if meson.version().version_compare('<0.62')
+metainfo_i18n = i18n.merge_file (
+    input:  metainfo_with_relinfo,
+    output: 'org.freedesktop.appstream.cli.metainfo.xml',
+    type: 'xml',
+    data_dirs: [meson.current_source_dir()],
+    po_dir: join_paths (source_root, 'po'),
+    install: true,
+    install_dir: metainfo_dir
+)
+else
 metainfo_i18n = i18n.itstool_join(
     input:  metainfo_with_relinfo,
     output: 'org.freedesktop.appstream.cli.metainfo.xml',
@@ -32,6 +43,7 @@ metainfo_i18n = i18n.itstool_join(
     install: true,
     install_dir: metainfo_dir,
 )
+endif
 
 test('as-validate_metainfo.cli',
     ascli_exe,
diff --git a/meson.build b/meson.build
index ce84a77..e378bad 100644
--- a/meson.build
+++ b/meson.build
@@ -1,5 +1,5 @@
 project('AppStream', 'c',
-  meson_version: '>=0.62',
+  meson_version: '>=0.59',
   default_options: ['c_std=gnu17', 'cpp_std=c++20'],
 
   license: 'LGPL-2.1-or-later',
@@ -73,6 +73,7 @@ if get_option('maintainer')
                          '-Winline',
                          '-Wno-error=comment',
                          '-Werror=format-security',
+                         '-Werror=format',
 
                          # used in rsvg's headers, so we can not make this fatal yet
                          '-Wno-error=expansion-to-defined',
diff --git a/po/meson.build b/po/meson.build
index e9ede19..f7c1d84 100644
--- a/po/meson.build
+++ b/po/meson.build
@@ -1,5 +1,19 @@
 
 as_gettext_domain = 'appstream'
+if meson.version().version_compare('<0.62')
+i18n.gettext(as_gettext_domain,
+    preset : 'glib',
+    data_dirs: [join_paths(source_root, 'data')],
+    args: [
+        '--default-domain=' + as_gettext_domain,
+        '--from-code=UTF-8',
+        '-i', '-F', '-c', '--no-wrap',
+        '--package-name=' + as_gettext_domain,
+        '--copyright-holder=Matthias Klumpp',
+        '--msgid-bugs-address=appstream@lists.freedesktop.org'
+    ]
+)
+else
 i18n_result = i18n.gettext(as_gettext_domain,
     preset : 'glib',
     data_dirs: [join_paths(source_root, 'data')],
@@ -12,6 +26,7 @@ i18n_result = i18n.gettext(as_gettext_domain,
         '--msgid-bugs-address=appstream@lists.freedesktop.org'
     ]
 )
+endif
 
 run_target ('make-linguas',
     command: ['sh',
-- 
2.49.0
